S/SwiftShader/build_tarballs.jl

```diff
@@ -2,37 +2,85 @@ using BinaryBuilder
+version = v"0.1.0" # there are no official versions yet
+source = "https://github.com/google/swiftshader.git"
+commit = "6c1ab2e3638260721c19b33017925f6deb9e30ac" # March 14th, 2023
 
 sources = [
-    ArchiveSource("https://github.com/google/gfbuild-swiftshader/releases/download/github%2Fgoogle%2Fgfbuild-swiftshader%2F$(commit)/gfbuild-swiftshader-$(commit)-Linux_x64_Release.zip",
-                  "20c0bba105e30b1e5ddfa73c69c2400a615bf2e76d124330897226d2fa512a50"; unpack_target="x86_64-linux-gnu"),
-    ArchiveSource("https://github.com/google/gfbuild-swiftshader/releases/download/github%2Fgoogle%2Fgfbuild-swiftshader%2F$(commit)/gfbuild-swiftshader-$(commit)-Mac_x64_Release.zip",
-                  "8078ef21ef02aa9f9bcf69ed303ed28b0cea1feb3129a9196385b5cd0b378c28"; unpack_target="x86_64-apple-darwin14"),
-    ArchiveSource("https://github.com/google/gfbuild-swiftshader/releases/download/github%2Fgoogle%2Fgfbuild-swiftshader%2F$(commit)/gfbuild-swiftshader-$(commit)-Windows_x64_Release.zip",
-                  "22354a01605adb3310736fc85cd1e20039bd4532f428226f8b3ea60d0f2bf853"; unpack_target="x86_64-w64-mingw32"),
+    GitSource(source, commit),
+    DirectorySource("./patches"),
 ]
 
 script = raw"""
-cd $WORKSPACE/srcdir
-mkdir -p "${libdir}"
-mv ${target}/lib/* ${libdir}/.
-rm -rf ${libdir}/*version
-install_license ${target}/OPEN_SOURCE_LICENSES.TXT
+cd swiftshader
+
+# Remove architecture-specific flags.
+atomic_patch -p1 ${WORKSPACE}/srcdir/remove_march.patch
+
+CXX_FLAGS=()
+CMAKE_FLAGS=()
+
+# Release build for best performance
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
+
+# Install things into $prefix
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+
+# Explicitly use our cmake toolchain file
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+
+# Work around undefined `cinttypes` printing macros.
+CXX_FLAGS+="-D__STDC_FORMAT_MACROS"
+
+# Do not enable the -Werror flag (builds fine without it).
+CMAKE_FLAGS+=(-DSWIFTSHADER_WARNINGS_AS_ERRORS=FALSE)
+
+# Don't build unit tests (they currently error the build because of yet unindentified libc++ issues).
+CMAKE_FLAGS+=(-DSWIFTSHADER_BUILD_TESTS=FALSE)
+
+# Handle Mac SDK <10.12 errors.
+if [[ "${target}" == *-apple-* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=10.12
+    CMAKE_FLAGS+=(-DLLVM_ENABLE_LIBCXX=ON)
+fi
+
+if [[ "${target}" == *-mingw* ]]; then
+    find . -type f -exec sed -i 's/include <Windows\.h>/include <windows\.h>/g' {} +
+    CXX_FLAGS+=" -DHAVE_UNISTD"
+fi
+
+cmake -B build -S . -GNinja ${CMAKE_FLAGS[@]} -DCMAKE_CXX_FLAGS="${CXX_FLAGS}"
+ninja -C build -j ${nproc} install
+
+folderName=""
+if [[ "${target}" == *-apple-* ]]; then
+    echo "hello"
+    folderName=Darwin
+elif [[ "${target}" == *-linux-* ]]; then
+    folderName=Linux
+elif [[ "${target}" == *-mingw* ]]; then
+    folderName=Windows
+fi
+
+mv build/$folderName/* ${libdir}
+
+install_license LICENSE.txt
 """
 
 platforms = [
```

Comment:
Quoting [from the log](https://buildkite.com/julialang/yggdrasil/builds/1535#0186f0e8-d183-4446-b37e-dfc23d77c295/638-2251):
```
┌ Warning: /cache/build/yggy-amdci7-6/julialang/yggdrasil/S/SwiftShader/build/x86_64-linux-gnu/LZPx879y/x86_64-linux-gnu-libgfortran5-cxx11/destdir/lib/libSPIRV-Tools-shared.so contains std::string values! This causes incompatibilities across the GCC 4/5 version boundary. To remedy this, you must build a tarball for both GCC 4 and GCC 5. To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│
│  platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /cache/julia-buildkite-plugin/depots/e2fd9734-29d8-45cd-b0eb-59f7104f3131/packages/BinaryBuilder/ejTbj/src/auditor/compiler_abi.jl:247
```

Reply:
Is it necessary with a GCC version > 5?

Reply:
Yes.